pkg/controllers/workapplier/controller.go

```diff
@@ -196,6 +194,27 @@ func (h *PriorityQueueEventHandler) Generic(ctx context.Context, evt event.Typed
 	h.AddToPriorityQueue(ctx, evt.Object, false)
 }
 
+var defaultRequeueRateLimiter *RequeueFastSlowWithExponentialBackoffRateLimiter = NewRequeueFastSlowWithExponentialBackoffRateLimiter(
+	// Allow fast requeues for 20 attempts (5 seconds each, 100 seconds of fast requeues).
+	20,
+	// Important (chenyu1): before the introduction of the requeue rate limiter, the work
+	// applier uses static requeue intervals, specifically 5 seconds (if the work object is unavailable),
+	// and 15 seconds (if the work object is available). There are a number of test cases that
+	// implicitly assume this behavior (e.g., a test case might expect that the availability check completes
+	// w/in 10 seconds), which is why the rate limiter uses the 5 seconds fast requeue delay by default.
+	// If you need to change this value and see that many test cases begin to fail, update the test
+	// cases accordingly.
+	5,
+	// Then switch to exponential backoff with a base of 1.2 with a cap of 5 minutes.
+	1.2,
+	300,
```

Comment:
this is probably still a bit too aggressive

Reply:
per earlier discussion, the default max. delay has been set to 900 secs (15 minutes). The fast backoff factor is now 1.5 instead of 1.2.